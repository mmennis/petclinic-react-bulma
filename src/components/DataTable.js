import React from 'react';
import { Component } from 'react';
import PropTypes from 'prop-types';

export class TableHeader extends Component {

    constructor(props) {
        super(props);
        this.headerList = props.headers;
        this.createHeader = this.createHeader.bind(this);
    }

    render() {
        return (
            <thead>
                <tr>
                    {
                        this.createHeader(this.headerList)
                    }
                </tr>
            </thead>
        )
    }

    createHeader(headerList) {
        let table = [];
        headerList.map((header, i) => {
            table.push(
                <th key={i}>{header}</th>
            );
        })
        return table;
    }
}

TableHeader.propTypes = {
    headers: PropTypes.array.isRequired
}