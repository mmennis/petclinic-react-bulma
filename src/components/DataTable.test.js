import React from 'react';
import { TableHeader } from './DataTable';
import { create } from 'react-test-renderer';
import { render } from '@testing-library/react';


describe ('DataTableModule', () => {

    describe('TableHeader', () => {
        it('should render a table header per snapshot', () => {
            const tableHeader = create(<TableHeader headers={['one', 'two', 'three', 'four']} />);
            expect(tableHeader.toJSON()).toMatchSnapshot();
        })

        it('should have the headers in the table headers', () => {
            console.error = jest.fn();
            const { getByText } = render(<TableHeader headers={['testheader']}/>);
            const header = getByText(/testheader/i);
            expect(header).toBeInTheDocument();
            expect(console.error).toHaveBeenCalledTimes(1); // getByText logs an error
        })

        it('should fail to work if headers is missing', () => {
            console.error = jest.fn();
            React.createElement(TableHeader);
            expect(console.error).toHaveBeenCalledTimes(1);
        })
    })


})