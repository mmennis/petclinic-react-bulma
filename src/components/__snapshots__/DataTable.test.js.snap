// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTableModule TableHeader should render a table header per snapshot 1`] = `
<thead>
  <tr>
    <th>
      one
    </th>
    <th>
      two
    </th>
    <th>
      three
    </th>
    <th>
      four
    </th>
  </tr>
</thead>
`;
